testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jsf/testurl/ForbiddenUrlTestCase.java

```diff
@@ -98,13 +100,32 @@ public void testAllowedUrl() throws Exception {
     public void testForbiddenUrl() throws Exception {
         HttpClientBuilder httpClientBuilder = HttpClientBuilder.create();
 
-        try(CloseableHttpClient client = httpClientBuilder.build()) {
+        try (CloseableHttpClient client = httpClientBuilder.build()) {
 
-            HttpUriRequest getVarRequest = new HttpGet(url.toExternalForm() + FORBIDDEN_URL);
+            HttpUriRequest getVarRequest = new HttpGet(url.toExternalForm() + jakartafiUrl(FORBIDDEN_URL));
             try (CloseableHttpResponse getVarResponse = client.execute(getVarRequest)) {
                 int statusCode = getVarResponse.getStatusLine().getStatusCode();
-                Assert.assertEquals("Status code should be " + FORBIDDEN_STATUS_CODE + ", but is " + statusCode + ". For more information see JBEAP-18842. ", FORBIDDEN_STATUS_CODE, statusCode);
+                Assert.assertEquals("Status code should be " + FORBIDDEN_STATUS_CODE + ", but is " + statusCode +
+                        ". For more information see JBEAP-18842. ", FORBIDDEN_STATUS_CODE, statusCode);
             }
         }
     }
+
+    /**
+     * TODO: Temporary hack
+     * EE 9 changes the namespace from javax.* to jakarta.*. Running these tests under EE 9 or later, as WildFly Preview does, will
+     * cause the tests to fail, since they're using the old namespace in strings. This method will alter the URLs if it can detect
+     * that the tests are running under EE 9 or later. Once WildFly switches completely to EE 10, this hack can be removed, and the
+     * strings updated appropriately.
+     * @param url
+     * @return
+     */
+    @Deprecated
+    private String jakartafiUrl(String url) {
+        if (System.getProperty("ts.ee9") != null) {
```

Comment:
Instead of checking the active profile, which will break for other profiles or in big-bang, please inspect the FQCN of a Faces API class. If it starts with 'jakarta' then the test is in a 'jakarta' environment.

Reply:
This is a client test (@RunAsClient), so the namespace is not being transformed, giving me a javax.faces.context.FacesContext in the test method. Class.forName("jakarta.faces.context.FacesContext") throws a CNFE. Am I missing a trick?

Reply:
Ah, ok, sorry; then testing for the profile is ok. Please use AssumeTestGroupUtil.isWildFlyPreview() though.

#15725 can back this out, as the class already uses the jakarta.* URL in the big-bang branch. That's just minor housekeeping though as the method will still work, it just won't do anything.